### PR TITLE
Interface to UI for UKy

### PR DESCRIPTION
**Pull request by dangunter**
## Summary/Motivation:
Shows the feasibility of using the WaterTAP UI for PROMMIS models.

## Changes proposed in this PR:
- Add UI module, diagram, and some CSV exports in the `prommis.uky` subpackage
- At runtime, load this information into the WaterTAP UI

## Dependencies

https://github.com/watertap-org/watertap/pull/1292

## Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.md file
   at the top level of this directory.

2. I represent I am authorized to make the contributions and grant the license. If my employer has
   rights to intellectual property that includes these contributions, I represent that I have
   received permission to make contributions and grant the required license on behalf of that
   employer.

### CODE REVIEW

**Comment by dangunter** [comment]
Currently waiting for #25 to resolve.

File: src/prommis/uky/tests/test_uky_flowsheet_ui.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import pytest
+from prommis.uky import uky_flowsheet_ui as ui
```
Comment (lbianchi-lbl) [inline src/prommis/uky/tests/test_uky_flowsheet_ui.py:2]:
```suggestion
import pytest
ui = pytest.importorskip("prommis.uky.uky_flowsheet_ui", reason="Flowsheet UI components not available")
```

**Review by lbianchi-lbl — CHANGES_REQUESTED** [review]
Needs `pytest.importorskip` so that the tests pass when WaterTAP (which we're treating as an optional dependency) is not installed.

**Comment by dangunter** [comment]
This PR seems to be stuck due to a codecov error that has these characteristics:
* Happens every time
* Only occurs with the Linux / Python 3.11 / Dev workflow

No idea what to do next, except force merge?
Thoughts, @lbianchi-lbl?

**Comment by dangunter** [comment]
OK, it works now. We should approve it quick before it breaks again.

**Review by bpaul4 — COMMENTED** [review]
@dangunter the code all works and the new content looks good, but I'm not sure how to generate the PNG file (or should it open a browser tab?). It would also be helpful if you could add some light documentation.

**Review by MarcusHolly — APPROVED** [review]
LGTM

**Comment by dangunter** [comment]
@bpaul4 good point about the documentation, I will work on that.

I am not sure I understand the PNG question. Perhaps you are assuming it is autogenerated; it is just a picture. It should display in the initial "input" screen when that model is loaded into the WaterTAP UI.

**Review by bpaul4 — APPROVED** [review]
I'll trust that this works as intended, given that the tests are passing and everything looks good.

**Pull request merged**